window.docContents[143].resolve({"/#PSBP-Documentation--Programming-With-Failure--safeDivIsOne-next":{"id":"/#PSBP-Documentation--Programming-With-Failure--safeDivIsOne-next","header":"10.6. safeDivIsOne","context":"PSBP Documentation\u0009Programming With Failure","contents":"Let's see how failures accumulate\n\ndef twiceSafeDiv\n[Functional program]\n    [Creational program]\n    [Sequential program]\n    [Conditional program]\n    [WithFailure String program] :\n  program ((Nat × Nat) × Nat) Nat :=\n    (first >=> safeDiv) &&& second >=> safeDiv\n\n\n#eval\n  materializeActiveWithFailure\n    twiceSafeDiv\n    ((10, 5), 2)\n\n\nSum.inr 1\n\n\n#eval\n  materializeActiveWithFailure\n    twiceSafeDiv\n    ((10, 2), 0)\n\n\nSum.inl \"tried to divide 5 by 0\"\n\n\n#eval\n  materializeActiveWithFailure\n    twiceSafeDiv\n    ((10, 0), 2)\n\n\nSum.inl \"tried to divide 10 by 0\"\n\n\nThe last example shows the fail-fast behavior of failureWith.\n\n\n"}});